modules/ballerina-core/src/main/java/org/ballerinalang/util/codegen/CodeGenerator.java

```diff
@@ -670,7 +670,15 @@ public void visit(ParameterDef parameterDef) {
 
     @Override
     public void visit(VariableDef variableDef) {
-
+        MemoryLocation memoryLocation = variableDef.getMemoryLocation();
+        if (memoryLocation instanceof StackVarLocation || memoryLocation instanceof WorkerVarLocation) {
+            int lvIndex = getNextIndex(variableDef.getType().getTag(), lvIndexes);
```

Comment:
This code block and the code block in 'visit(VariableDefStmt varDefStmt)' can be merged.